Feature/lib 24

Lelik7777 opened the pull request:
# Checkbox Component Implementation

## Overview
This pull request introduces a new customizable Checkbox component using Radix UI, along with necessary style updates and improved project structure.

## Key Changes

### UI Component
- Implemented a new Checkbox component using @radix-ui/react-checkbox
- Added customizable props: checked, disabled, label
- Implemented forwarded ref for better integration
- Created comprehensive Storybook stories showcasing various states and behaviors

### Styling
- Added SCSS modules for Checkbox styling with support for different states
- Created _fonts.scss file for font imports (Inter weights 400, 500, 600, 700)
- Updated index.scss to forward all style modules including fonts
- Improved overall style organization and modularity

### Project Structure and Configuration
- Updated vite.config.ts: uncommented rollupTypes
- Refactored create-component extension from .cjs to .mjs
- Updated package.json scripts for better documentation and usage

## Features
- Checkbox supports checked, unchecked, and disabled states
- Custom labels can be added to checkboxes
- Both controlled and uncontrolled usage is supported
- Comprehensive Storybook documentation for easy component exploration

## Testing
- Storybook stories cover all major use cases and states of the Checkbox component

## Next Steps
- Consider adding unit tests for the Checkbox component
- Explore integration with form libraries if needed
- Gather feedback on the component's API and styling

This pull request provides a fully functional and well-documented Checkbox component, ready for use in the project, with improved styling and project structure.

Lelik7777 committed:
"feat(ui): add Radix UI Checkbox component

- Install @radix-ui/react-checkbox package
- Add comments to package.json scripts for better documentation
- Update scripts section with descriptions"
- Update vite.config.ts: uncomment  rollupTypes

Lelik7777 committed:
"feat(styles): add font imports and reorganize style structure

- Create _fonts.scss file for font imports
- Import Inter font weights 400, 500, 600, and 700
- Update index.scss to forward all style modules including fonts
- Improve overall style organization and modularity"

Lelik7777 committed:
feature(checkbox):add checkbox folder with templates

 - Refactor: change extension create-component from cjs to mjs
 - Update package.json: script "comp"

Lelik7777 committed:
feat(Checkbox): implement customizable Checkbox component

- Create Checkbox component with customizable props (checked, disabled, label)
- Implement forwarded ref for better integration
- Add SCSS modules for styling with support for different states
- Create Storybook stories to showcase various Checkbox states and behaviors

This commit provides a fully functional and well-documented Checkbox component
ready for use in the project, with comprehensive Storybook documentation.

EvgeniyaYakunina commented on src/components/ui/checkbox/checkbox.stories.tsx:
```diff
@@ -12,6 +16,55 @@ const meta = {
+}
+export const ControlCheckbox: Story = {
+  render: args => {
+    const [checked, setChecked] = React.useState(false)
```
Тут я бы не писала React  перед useState. По идее тут он никакой роли не играет.

Lelik7777 replied:
Мы таким образом экономим на импорте. Я где-то сталкивался с этой практикой, когда импользуем хуки

EvgeniyaYakunina commented on src/components/ui/checkbox/checkbox.stories.tsx:
```diff
@@ -12,6 +16,55 @@ const meta = {
+    label: 'controlled  checkbox',
+  },
+  render: args => {
+    const [checked, setChecked] = React.useState(false)
```
Тут тоже React  не нужен

EvgeniyaYakunina commented on src/components/ui/checkbox/checkbox.tsx:
```diff
@@ -1,13 +1,74 @@
+import { CheckmarkOutline } from '@/assets'
 
-export type CheckboxProps = {}
+type CheckboxProps = {
+  checked?: boolean
+  className?: string
+  disabled?: boolean
+  id?: string
+  label?: string | null
+  onChange?: (checked: boolean) => void
+}
```
Тут в типизации у тебя должен быть только label, остальное должно из коробки уже быть в компоненте радикса.

Lelik7777 replied:
fixed

EvgeniyaYakunina commented on src/components/ui/checkbox/checkbox.tsx:
```diff
@@ -1,13 +1,74 @@
 
-  return <div className={clsx(classNames.root)}></div>
-}
+export const Checkbox = forwardRef<ElementRef<typeof CheckboxRadix.Root>, CheckboxProps>(
+  ({ checked, disabled, id, label, onChange }: CheckboxProps, ref) => {
```
Тут у тебя типизация CheckboxProps два раза прописана. Во втором случае ее можно убрать.

Lelik7777 replied:
fixed

EvgeniyaYakunina commented on src/components/ui/checkbox/checkbox.tsx:
```diff
@@ -1,13 +1,74 @@
 
-  return <div className={clsx(classNames.root)}></div>
-}
+export const Checkbox = forwardRef<ElementRef<typeof CheckboxRadix.Root>, CheckboxProps>(
+  ({ checked, disabled, id, label, onChange }: CheckboxProps, ref) => {
```
И еще, мне кажется, надо добавить {...restProps} на случай если появятся еще какие-то пропсы,которые нам надо будет прокинуть.

Lelik7777 replied:
fixed

EvgeniyaYakunina reviewed (commented):
Алексей, посмотри, пожалуйста, мои комментарии. Если не согласен с чем-то, то предлагаю обсудить.

Lelik7777 committed:
refactor(Checkbox): update props and state handling

- Update CheckboxProps type to include Radix UI props
- Modify onChange prop to handle CheckboxRadix.CheckedState
- Update component to handle 'indeterminate' state
- Adjust useState initial value in stories to match new type

This commit improves type safety and aligns the Checkbox component
more closely with Radix UI's API, while maintaining backwards
compatibility with boolean checked states.
